samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -42,8 +42,8 @@ func (a PetApi) AddPet (body Pet) (APIResponse, error) {
   path := a.Configuration.BasePath + "/pet"
 
   // verify the required parameter 'body' is set
-  if &body == nil {
-      return *NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
+  if reflect.DeepEqual((Pet{}), body) {
```

Comment:
@guohuang Thinking about this even further... just because a parameter is required doesn't mean it can't be the zero value. E.g. with `Pet` above, I think it would be legal for all the fields of `Pet` to be zero values (e.g. empty strings, etc). So this code is problematic.